packages/core/src/internal-components/popup/popup.element.ts

```diff
@@ -0,0 +1,273 @@
+ * ```typescript
+ * import '@cds/core/internal-components/popup/register.js';
+ * ```
+ *
+ * ```html
+ * <cds-internal-popup>
+ *  <section cds-layout="vertical align:horizontal-stretch">
+ *    <div cds-layout="vertical pad:md gap:md">
+ *      <h2 cds-text="display">A Title</h2>
+ *      <div>
+ *        <p cds-text="body">
+ *          Content inside a generic popup.
+ *        </p>
+ *      </div>
+ *    </div>
+ *  </section>
+ * </cds-internal-popup>
+ * ```
+ *
+ * @beta
+ * @element cds-internal-popup
+ * @slot - Content slot for the content inside the popup's panel
+ * @event closeChange - Notify user when close event has occurred
+ * @cssprop --active-corner-border-radius
+ * @cssprop --background
+ * @cssprop --backdrop-background
+ * @cssprop --layered-backdrop-background
+ * @cssprop --border-color
+ * @cssprop --border-radius
+ * @cssprop --border-width
+ * @cssprop --box-shadow
+ * @cssprop --close-button-offset
+ * @cssprop --color
+ * @cssprop --min-height
+ * @cssprop --min-width
+ * @cssprop --max-height
+ * @cssprop --max-width
+ * @cssprop --mobile-max-height
+ * @cssprop --overflow - sets overflow x and y values respectively
+ * @cssprop --height
+ * @cssprop --width
+ * @cssprop --animation-duration
+ * @cssprop --animation-easing
+ *
+ * KNOWN ISSUE: Safari jumps through the exit animation but only when the ESC key is pressed.
+ *
+ */
+@animate({
+  hidden: {
+    true: reverseAnimation(AnimationResponsivePopupEnterName),
+    false: AnimationResponsivePopupEnterName,
+  },
+})
+export class CdsInternalPopup extends CdsInternalStaticOverlay implements Animatable {
+  // --- mixins/composables ---
+
+  protected ariaPopupController = new AriaPopupController(this);
+
+  @i18n() i18n = I18nService.keys.popup;
+
+  @property({ type: String })
+  cdsMotion = 'on';
+
+  /**
+   * force responsive by setting orientation to 'none'.
+   * do not manipulate the responsive property because the
+   * positioning logic may override your changes.
+   */
+  /** @private */
+  @property({ type: Boolean, reflect: true })
+  responsive = false;
+
+  @event()
+  cdsMotionChange: EventEmitter<string>;
+
+  // --- anchor ---
+
+  /**
+   *
+   * @type {HTMLElement}
+   * @memberof CdsInternalPopup
+   * Used to perform lookups for anchors. If unset, it assumes that the document is the container
+   * and will then only check for anchors in the light DOM. If set to something other than the
+   * document, it will perform checks in both the light and shadow DOM of the given container.
+   * This is useful in the situation where an anchor is wrapped up with the dropdown in a parent
+   * wrapper element.
+   *
+   */
+  container: HTMLElement;
+
+  private anchorId = '';
+
+  /** @private */
+  anchorElement: HTMLElement | null = null;
+
+  /**
+   * Allows an element id to be passed through an HTML attribute in order to set the anchor element
+   */
+  @property({ type: String, reflect: true })
+  get anchor() {
+    return this.anchorId;
```

Comment:
Done. I kept the `get anchorElement` as only private from the docs because I use it to test the anchor logic in isolation. Tests run. I will need to verify with manual testing though too.